packages/playground-server/src/node.ts

```diff
@@ -40,30 +50,86 @@ export class LocalFirebaseServiceFactory extends FirebaseServiceFactory {
   }
 }
 
-export let serviceFactory: FirebaseServiceFactory;
+class StandardFirebaseServiceFactory extends FirebaseServiceFactory
+  implements IClosableFirebaseServiceFactory {
+  constructor(params: any) {
+    super(params);
+  }
+  async closeServiceConnections() {}
+}
+
+class LocalPersistentFirebaseServiceFactory extends FirebaseServiceFactory
+  implements IClosableFirebaseServiceFactory {
+  firebaseServer: FirebaseServer;
+  constructor(
+    private readonly host: string,
+    private readonly port: string,
+    snapshot: any
+  ) {
+    super({
+      databaseURL: `ws://${host}:${port}`,
+      projectId: "projectId",
+      apiKey: "",
+      authDomain: "",
+      storageBucket: "",
+      messagingSenderId: ""
+    });
+
+    this.firebaseServer = new FirebaseServer(this.port, this.host, snapshot);
+  }
+
+  static async create(host: string, port: string) {
+    const snapshot = await getPlaygroundSnapshot();
+
+    return new LocalPersistentFirebaseServiceFactory(host, port, snapshot);
+  }
+
+  async closeServiceConnections() {
+    const snapshot = await this.firebaseServer.getValue();
+    storePlaygroundSnapshot(snapshot);
+
+    await this.firebaseServer.close();
+  }
+}
+
+export let serviceFactory: Promise<IClosableFirebaseServiceFactory>;
 
 console.log(`Using Firebase configuration for ${process.env.NODE_ENV}`);
 if (!devAndTestingEnvironments.has(process.env.NODE_ENV!)) {
   confirmFirebaseConfigurationEnvVars();
-  serviceFactory = new FirebaseServiceFactory({
-    apiKey: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.apiKey]!,
-    authDomain: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.authDomain]!,
-    databaseURL: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.databaseURL]!,
-    projectId: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.projectId]!,
-    storageBucket: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.storageBucket]!,
-    messagingSenderId: process.env[
-      FIREBASE_CONFIGURATION_ENV_KEYS.messagingSenderId
-    ]!
-  });
+  serviceFactory = Promise.resolve(
+    new StandardFirebaseServiceFactory({
+      apiKey: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.apiKey]!,
+      authDomain: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.authDomain]!,
+      databaseURL: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.databaseURL]!,
+      projectId: process.env[FIREBASE_CONFIGURATION_ENV_KEYS.projectId]!,
+      storageBucket: process.env[
+        FIREBASE_CONFIGURATION_ENV_KEYS.storageBucket
+      ]!,
+      messagingSenderId: process.env[
+        FIREBASE_CONFIGURATION_ENV_KEYS.messagingSenderId
+      ]!
+    })
+  );
 } else {
   confirmLocalFirebaseConfigurationEnvVars();
-  serviceFactory = new LocalFirebaseServiceFactory(
-    process.env.FIREBASE_SERVER_HOST!,
-    process.env.FIREBASE_SERVER_PORT!
-  );
+  if (process.env.PLAYGROUND_PERSISTENCE_ENABLED) {
```

Comment:
It's probably a good idea to mention the existence of this env var and its function as part of the README 

(similar to how you've described this feature in the OP)
> In order to enable the feature the env parameter PLAYGROUND_PERSISTENCE_ENABLED=true has to be set.